### ISSUE TITLE
Enforce strict mode on ES6 modules when not implicit due to transpiling issue

### ISSUE THREAD

**Issue by AnthonyAstige** [dferber90/eslint-plugin-meteor#208]
This is a feature request.

[Meteor modules should but don't execute in strict mode](https://github.com/meteor/meteor/issues/7050), which [makes ESLint unhappy](https://forums.meteor.com/t/eslint-for-meteor-1-3-modules-and-strict-mode/23105/4).  It's [doubtful ESLint or Meteor will patch soon, but we could fork the ESLint strict rule to work for Meteor now](https://github.com/eslint/eslint/issues/6184).

If I coded up something for this, might you accept a pull request?

Note: This would be my first ESLint plugin coding.

**Comment by dferber90** [comment]
> If I coded up something for this, might you accept a pull request?

I will :)

What exactly are you going to create? A modified version of `strict` which does not warn about this in modules?

```
'use strict' is unnecessary inside of modules. (strict)
```

It's possible to change the ESLint settings on a per-folder basis by adding a `eslintrc` file into the folder. Would there be a simpler way of just disabling the rule for that folder? But then we'd lose the other warnings it generates for them. So probably not, right?

> Note: This would be my first ESLint plugin coding.

[This](https://github.com/dferber90/eslint-plugin-meteor/blob/master/docs/guides/development.md) should be a good starting point. I'm not sure it is 100% up-to-date though.

Feel free to open an issue if you need help!


---

Some starting tips:
If I understood correctly, you're going to create a modified version of the `strict` rule.
I think you'll need to determine whether it's a module or not by looking at the file path, then execute either the default rule or the modified rule.

Make sure the rule behaves like the original in non-modules. This is kind of annoying as we'd have to keep up with changes to the original implementation. However it's only temporary (until Meteor is compliant), so I'm okay with it.

You should add the rule to the ESLint recommended settings exported by the plugin in `index.js` then, and make sure test coverage stays at 100% :)

**Comment by AnthonyAstige** [comment]
@dferber90 Awesome, thanks for the help in getting started here :)

Yeah you got it exactly why this rule is needed - The strict rule can be turned off to get rid of that `...unnecessary inside..` error, but then I miss the main point, enforcing strict mode everywhere.  Per folder doesn't help, as I need a strict rule everywhere.

P.S. I'm [discussing with Meteor](https://github.com/meteor/meteor/issues/7050#issuecomment-219444342) for a more direct solution before diving into code here just in case

> ... that still leaves having to place `'use strict;'` on the top of potentially hundreds of  `/import/**.js` files.

> Before I delve into making lint rule, I wonder if it's possible to create an npm package to address this. It'd be nicer to do something like meteor npm install meteor-strict-imports --save. Is there anything in the Meteor core which a package could hook into to achieve this?

**Comment by AnthonyAstige** [comment]
Looks like a work-around via .babelrc is [coming in Meteor 1.3.3](https://github.com/meteor/meteor/issues/7050#issuecomment-220667308).  I'm gonna close this issue since there will be a much cleaner solution soon enough.

@dferber90 Thanks again for the response, I'll keep everything in mind in case I need to build a different meteor linting rule.

**Comment by dferber90** [comment]
No worries, glad it gets a proper solution. Thanks for closing :)
